server/src/lib/clip.ts

```
@@ -120,64 +120,71 @@ export default class Clip {
+      } else {
+        // For non-base64 uploads, we can just stream data.
+        transcoder = new Transcoder(request);
+      }
 
-    await Promise.all([
-      this.s3
-        .upload({
-          Bucket: getConfig().BUCKET_NAME,
-          Key: clipFileName,
-          Body: transcoder
-            .audioCodec('mp3')
-            .format('mp3')
-            .stream(),
-        })
-        .promise(),
-      this.s3
-        .putObject({
-          Bucket: getConfig().BUCKET_NAME,
-          Key: sentenceFileName,
-          Body: sentence,
-        })
-        .promise(),
-    ]);
-
-    console.log('file written to s3', clipFileName);
-
-    await this.model.saveClip({
-      client_id: client_id,
-      locale: params.locale,
-      original_sentence_id: filePrefix,
-      path: clipFileName,
-      sentence,
-      sentenceId: headers.sentence_id,
-    });
-    await Awards.checkProgress(client_id);
+      await Promise.all([
+        this.s3
+          .upload({
+            Bucket: getConfig().BUCKET_NAME,
+            Key: clipFileName,
+            Body: transcoder
+              .audioCodec('mp3')
+              .format('mp3')
+              .stream(),
+          })
+          .promise(),
+        this.s3
+          .putObject({
+            Bucket: getConfig().BUCKET_NAME,
+            Key: sentenceFileName,
+            Body: sentence,
+          })
+          .promise(),
+      ]);
+
+      console.log('file written to s3', clipFileName);
+
+      await this.model.saveClip({
+        client_id: client_id,
+        locale: params.locale,
+        original_sentence_id: filePrefix,
+        path: clipFileName,
+        sentence,
+        sentenceId: headers.sentence_id,
+      });
+      await Awards.checkProgress(client_id);
 
-    response.json(filePrefix);
+      response.json(filePrefix);
+    } catch (error) {
+      console.log(error);
+      response.status(error.statusCode || 500).json(error);
```

let's just return a generic error message here, error will be meaningless to the client anyway.

---

Ok, then just send some error message then in below speak.tsx only check for that error message to differentiate between the errors ??

---

let me do this

---

I think it'd be best if we just console.error'd it here and output something generic, as the client doesn't need to know about the internals here.

---

:+1: